src/url-params.ts

```diff
@@ -163,10 +163,20 @@ export class AnalyticsUrlParams {
 	/**
 	 * @return full query parameter string that can be appended to URLs
 	 */
-	getQueryString(): string {
-		return [this.getDeviceIdsQueryString(), this.getSessionIdQueryString()]
-			.filter(x => x)
-			.join('&');
+	getQueryString(destinationHostname?: string): string {
```

Comment:
Just thinking about the use case, what do we expect from the UI code?
If we're preparing to navigate somewhere maybe the full URL is already at hand?

What I want to say, it would be good to not require any additional (verbose) operations or the UI side, trying to figure out if extracting the hostname would count as one.

Reply:
Agreed with Pranas. It's simpler to expect the full destination URL here instead of hostname. I'd suggest to make this function work with any valid URL string.

Reply:
Right, I wrote it like this because Im not familiar to how each component handles this function. Hub for example was transforming a url string to a `URL` type for the base url, so If I wanted to use that, I would have to switch the type of the argument to URL. Then maybe marketing site uses just `window.location.href` and then It would need to transform it to a URL type to pass it and so. So I though that to make it explicit for the component to pass just the hostname would be best but Im not sure.

Reply:
Yeah, I can see how it depends on our component implementations.

However, even in different use cases, I'd expect these things:
- If URLs are at hand then maybe they can be casted easily, i.e. `String(window.location.href)` and so
- Usage to converge over time, i.e. components to start using similar means to navigate

Reply:
Since this is a new method and we'd like to roll it out ourselves, it's fine for us to dictate the format and type of the parameter here. We will update the relevant places on components to follow the practice we set. Then I don't have a strong opinion on expecting a URL object or URL string here. Since this is a Typescript project, maybe it's better to strictly use URL type.

https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules_typedoc_node_modules_typescript_lib_lib_dom_d_.url.html

Reply:
@gelbal @pranasziaukas  just to be clear, what I meant is that, at least on Hub for example, they dont hande the URL as a string, but they handle it as an URL object. So I have to use any of the string attributes of it, be it href, hostname, or whatever. While on Docs for example [on this line](https://github.com/balena-io/docs/blob/3183fdc0f98ea21487b86635dc0f693183461fa2/templates/_track.html#L38) they handle everything on the return explicitly for links that send to dashboard only, there is no logic for all external links, so the maintainer would have to either explicitly set an argument, either the href, hostname, whatever, or send the URL object, but not both.

Reply:
So if we built getQueryStrings to expect URL objects, then each maintainer would have to make the change to take the href and build the URL object. If we leave it as to expect a string, it would need to explicitly send href, hostname or whatever. Or I could write the function to expect both either a URL object or a string and proceed with the respective logic whatever the case and then let each maintainer to choose to sue whatever suits him best. Should we aiming for the last scenario?

Reply:
Now its prepared to recieve both destinationURL and currentURL, both as URL objects and parse them for the regex.